### ISSUE TITLE
Alternate name: service-runner

### ISSUE THREAD

**Issue by gwicke** [wikimedia/service-runner#3]
See the title. Discuss!

**Comment by Krenair** [comment]
What about a name that doesn't include 'oid'?

**Comment by gwicke** [comment]
Do you have ideas?

**Comment by d00rman** [comment]
I'd like to keep the current name. If we create a separate pkg/repo for *complete* service templating, then the current name is quite appropriate - it supervises forked child processes and provides them with logging and metrics facilities out of the box.

If, on the other hand, we go with supervisoid being the *de facto* service template, then something like *servicoid* (to keep the *oid* ball rolling) would be more appropriate.

**Comment by subbuss** [comment]
I would propose that the 'oid' rolling ball be stopped :) "oid" as a suffix is an useful aid to understanding in some situations ( http://dictionary.reference.com/browse/-oid ), but not convinced it applies in all scenarios like "mathoid" or "citoid" for example. supervisoid is definitely quite awkward.

**Comment by gwicke** [comment]
Proposals, please ;)

**Comment by gwicke** [comment]
Alright, how about `service-sitter` ?

**Comment by d00rman** [comment]
> Alright, how about `service-sitter` ?

:+1:

**Comment by gwicke** [comment]
Another one: `servish`

**Comment by hardikj** [comment]
how about sericevisor?

**Comment by gwicke** [comment]
> how about servicevisor?

Or `servisor`?

**Comment by hardikj** [comment]
> servisor
:+1:

**Comment by subbuss** [comment]
service-sitter or servisor or service-master or service-manager ... Depends on whether this is an actual monitoring service for other services or if this is a template for cluster management for services ... I am unclear what this is to suggest names.

**Comment by gwicke** [comment]
@subbuss, it's cluster + standard config format + standard args + standard logging + standard metrics reporting at this point. There is no automatic monitoring of services other than the ones it runs. It only sets up workers & re-spawns as usual. Could easily add memory monitoring for workers as well. 

The idea is that we can hook it up with `npm start -- -c /path/to/config.yaml` and use that to run any of the service modules.

Back to the names.. I think my favorites right now are 'servish' and 'servisor', in that order.

**Comment by gwicke** [comment]
Went ahead & renamed to 'servisor' for now. It's not final & easy to change. But, lets settle on something soon.

**Comment by gwicke** [comment]
Another option that's still available: `service-runner`.

**Comment by subbuss** [comment]
That is indeed a very fine option.

**Comment by gwicke** [comment]
@d00rman said on IRC that he'd prefer something with more spirit, like `servish`.

Man, naming is hard ;)

**Comment by cscott** [comment]
I don't have strong opinions, but I'll note that 'service-sitter' and
'service-runner' are both much more intelligible to non-fluent English
speakers.  (The first has a bit of 'spirit', since it is a play on
baby-sitter which is an English idiom which doesn't translate well.  But
it's kind of funny to think about 'service-sitter' sitting on a service.
So if I had to vote, I'd vote for "service-sitter" as "just the right about
of clever" without being needlessly obscure or hard to parse.)

On Mon, Feb 16, 2015 at 8:46 PM, Gabriel Wicke <notifications@github.com>
wrote:

> @d00rman <https://github.com/d00rman> said on IRC that he'd prefer
> something with more spirit, like servish.
>
> Man, naming is hard ;)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wikimedia/servisor/issues/3#issuecomment-74605196>.
>



-- 
                         ( http://cscott.net/ )

**Comment by gwicke** [comment]
Also still available: `service-boilerplate`

**Comment by subbuss** [comment]
I think I'll go back to my previous comment ... how do you see this repository being used while building a service? Say, if we were building Parsoid today .. how would I use this new repository? 

Do I clone this, rename it to parsoid, and then add a whole bunch of code to it? If so, "service-boilerplate" is the right name.

On the other hand, if this repository is going to be used to manage a cluster of node processes on a node, etc. and it simply takes a config file provided by Parsoid to find out the entry point for kicking off parsoid with the right opts, then "service-runner" (or manager/sitter/) is probably a good name for it. That said, you could still make a case of "service-boilerplate" but that reads like it is more of a service code template than something that will actually be deployed by itself. "servisor" is not bad but it is a tad awkward but maybe one could get used to it with time ...

FWIW, I personally think clarity is more important for this repo than clever names. Clever / cool / unique names can be left for the actual services being built.

**Comment by d00rman** [comment]
@subbuss This is supposed to be just a service worker spawner/manager. It will be integrated into a [service template repository](https://github.com/wikimedia/service-template-node) we have begun to build (note: currently it's empty). So, if you were to start building Parsoid from scratch, you would clone that repo, tweak its config and do your development. *servisor* (or *service-sitter* or *servish* or etc.) would be a somewhat invisible component, since the template would provide you a runner script out of the box.

**Comment by gwicke** [comment]
To add to what @d00rman said, it's a module that you can also just use without any template. The template mainly helps with the remaining boilerplate.

We just discussed the naming issue on IRC, and finally settled on the boring `service-runner`.

**Comment by d00rman** [comment]
For the record, the only reason I don't like `service-runner` is because it looks like a name I'd come up with, and I'm notoriously bad with such imaginative tasks :smile:
